app/src/main/java/com/postpc/tenq/ui/activities/RoomActivity.java

```diff
@@ -67,4 +79,30 @@ public boolean onMove(@NonNull @NotNull RecyclerView recyclerView, @NonNull @Not
 
         itemTouchHelper.attachToRecyclerView(binding.recyclerTracks);
     }
+
+    /** inflate the menu */
+    public boolean onCreateOptionsMenu( Menu menu ) {
+
+        getMenuInflater().inflate(R.menu.inside_room, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    /**  control operations fpr clicking on the action buttons */
+    public boolean onOptionsItemSelected( @NonNull MenuItem item ) {
+
+        switch (item.getItemId()){
+            case R.id.share: // for debugging use this room id: "yYQK9C19BGy8nt5C4zxY"
+                ShareAlertDialog.share(RoomActivity.this, room.getId());
```

Comment:
Passing a context to a static method is a sure recipe for a memory leak 😁 
Actually, my code needs this fix as well.
I think the best approach would be something like this:
1. Rename `ShareAlertDialog` to `RoomSharingDialogUtil`.
2. Rename its method `share` to `public static AlertDialog getShareDialogForRoom(Context context, String roomId) { ...`
3. Return the AlertDialog you create there now, and don't call `.show()` on it.
4. In the activity:
```
AlertDialog dialog = RoomSharingDialogUtil.getShareDialogForRoom(this, roomId);
if (!isFinishing() && !isDestroyed()) dialog.show(); // check the activity is still alive and the context didn't leak.
```